### PR TITLE
Introduce braceless with

### PR DESCRIPTION
**Pull request by UplinkCoder**

### CODE REVIEW

**Review by thewilsonator — CHANGES_REQUESTED** [review]
* Header
* Tests
* Changelog

File: src/dmd/dscope.d [hunk 1]
```diff
@@ -69,6 +69,8 @@ enum SCOPE
     // The following are mutually exclusive
     printf        = 0x4_0000, /// printf-style function
     scanf         = 0x8_0000, /// scanf-style function
+
+    doublepop     = 0x10_0000, /// for "scope-less" with statements
```
Comment (ibuclaw) [inline src/dmd/dscope.d:73]: I beg your pardon?

Comment (UplinkCoder) [reply src/dmd/dscope.d:73]: would you like another name?

Comment (ibuclaw) [reply src/dmd/dscope.d:73]: It has no relation with the change at all.

**Comment by RazvanN7** [comment]
@UplinkCoder how do we proceed here? This would most likely require a DIP.

**Comment by RazvanN7** [comment]
cc @WalterBright @atilaneves

**Comment by atilaneves** [comment]
I don't understand why this would be desired.

**Comment by 12345swordy** [comment]
why this feature isn't behind a preview switch?

**Comment by ljmf00** [comment]
> why this feature isn't behind a preview switch?

I guess we need to ask another question first: what is the rationale behind this feature? What is the use-case here?

```d
enum Foo {
    A,
    B,
    C
}

void main()
{
    with(Foo) Foo foo = A;
}
```

This is already possible, although useless. I don't understand what the language doesn't cover already.

**Comment by dkorpel** [comment]
> what is the rationale behind this feature? What is the use-case here?

This was discussed in the newsgroup:
https://forum.dlang.org/post/bwudoaknbiodwilvldxl@forum.dlang.org

**Comment by ljmf00** [comment]
> > what is the rationale behind this feature? What is the use-case here?
> 
> This was discussed in the newsgroup: [forum.dlang.org/post/bwudoaknbiodwilvldxl@forum.dlang.org](https://forum.dlang.org/post/bwudoaknbiodwilvldxl@forum.dlang.org)

Thanks for the information! I expected that kind of context in the PR description, but I understand now the point.

I have some strong opinions against the usage of `with`, since it can potentially lead to more unreadable code, depending on the situation, so I'm probably biased. I would like to see more examples of this to reconsider my opinion and also other opinions with points in favour.

I think adding braces in those situations can clarify this a bit more. I'm not inclined to agree with this, especially when the `with` statement will become the special case among `if`, `while`, `for`,... I understand the usefulness of this on enums, but allowing this generally on big functions with a good level of cyclomatic complexity can lead to confusion and braces can help a lot in those situations.

**Comment by atilaneves** [comment]
I took another look at the forum post motivating this, and removing a level of indentation when using `with` is not something I agree with.

**Pull request closed without merging**